src/engine/database/neo4j.rs

```diff
@@ -805,3 +810,19 @@ impl From<Value> for bolt_proto::value::Value {
         }
     }
 }
+
+// TODO: fix this (jeffrey: could use your advise)
```

Comment:
You're definitely up against the limit of not being able to define a trait implementation where both the trait and struct are imported from other crates.  One option is the way you've done it, though I would consider renaming things, as you're not really converting to a Warpgrapher Value -- you're converting to a HashMap from String to Warpgraher Values.

Your other option would be to `impl TryFrom<HashMap<String, bolt_proto::value::Value>> for Value` and return a Value::Map, from which you could then borrow or extract the HashMap<String, Value> as needed.  This latter option is probably what I'd go for.

Reply:
Implemented the latter suggestion.